src/http_api_handler.erl

```diff
@@ -109,3 +115,99 @@ p2m(Bin) ->
 		    Bin
 	    end
     end.
+
+exo_get_value(Name, Fun, AccIn) ->
+    case exometer:get_value(Name) of
+	{ok, Value} ->
+	    Fun(Value, AccIn);
+	{error,not_found} ->
+	    AccIn
+    end.
+
+exo_entry_to_map({Name, Type, enabled}, Metrics) ->
+    exo_entry_to_map(Name, {Name, Type}, Metrics).
+
+exo_entry_to_map([Path], {Name, Type}, Metrics) ->
+    exo_get_value(Name, fun(V, Acc) ->
+				Entry = maps:from_list(V),
+				Acc#{Path => Entry#{type => Type}}
+			end, Metrics);
+exo_entry_to_map([H|T], Metric, Metrics) ->
+    Entry = maps:get(H, Metrics, #{}),
+    Metrics#{H => exo_entry_to_map(T, Metric, Entry)}.
+
+exo_entry_to_list({Name, Type, enabled}, Metrics) ->
+    exo_get_value(Name, fun(V, Acc) -> [{Name, Type, V}|Acc] end, Metrics).
+
+metrics(Path, json) ->
+    Entries = lists:foldl(fun exo_entry_to_map/2, #{}, exometer:find_entries(Path)),
+    Metrics = lists:foldl(fun(M, A) -> maps:get(M, A) end, Entries, Path),
+    jsx:encode(Metrics);
+metrics(Path, prometheus) ->
+    Metrics = lists:foldl(fun exo_entry_to_list/2, [], exometer:find_entries(Path)),
+    prometheus_encode(Metrics).
+
+prometheus_encode(Metrics) ->
+    lists:foldl(fun prometheus_encode/2, [], Metrics).
+
+prometheus_encode(V= {Path, Type, DataPoints}, Acc) ->
+    Name = make_metric_name(Path),
+    Payload = [[<<"# TYPE ">>, Name, <<" ">>, map_type(Type), <<"\n">>] |
+               [[Name, map_datapoint(DPName), <<" ">>, ioize(Value), <<"\n">>]
+                || {DPName, Value} <- DataPoints, is_valid_datapoint(DPName)]],
+    Payload1 = maybe_add_sum(Name, DataPoints, Type, Payload),
+    [Payload1, <<"\n">> | Acc].
+
+ioize(Atom) when is_atom(Atom) ->
+    atom_to_binary(Atom, utf8);
+ioize(Number) when is_float(Number) ->
+    float_to_binary(Number, [{decimals, 4}]);
+ioize(Number) when is_integer(Number) ->
+    integer_to_binary(Number);
+ioize(Something) ->
+    Something.
+
+make_metric_name(Path) ->
+    NameList = lists:join($_, lists:map(fun ioize/1, Path)),
+    NameBin = iolist_to_binary(NameList),
+    re:replace(NameBin, "-|\\.", "_", [global, {return,binary}]).
+
+map_type(undefined)     -> <<"untyped">>;
+map_type(counter)       -> <<"gauge">>;
```

Comment:
That does not make sense for your metrics if your counters are monotonically increasing, e.g. you can take "counter" as Prometheus type too if thats the case. In my code I did not force this because exometer counters are not necessarily that way.